#! /bin/sh /usr/share/dpatch/dpatch-run
## 28blankerprg_callsemantics.dpatch by  <hesso@pool.math.tu-berlin.de>
##
## DP: Allow 'blankerprg' to be called without arguments (as the manpage states).

@DPATCH@
--- screen-4.0.3.orig/comm.c	2008-04-18 01:56:21.000000000 +0200
+++ screen-4.0.3/comm.c	2008-04-18 01:56:27.000000000 +0200
@@ -98,7 +98,7 @@
 #endif
   { "blanker",		NEED_DISPLAY|ARGS_0},
 #ifdef BLANKER_PRG
-  { "blankerprg",	ARGS_1|ARGS_ORMORE },
+  { "blankerprg",	ARGS_0|ARGS_ORMORE },
 #endif
   { "break",		NEED_FORE|ARGS_01 },
   { "breaktype",	NEED_FORE|ARGS_01 },
--- screen-4.0.3.orig/process.c	2008-04-18 02:10:35.000000000 +0200
+++ screen-4.0.3/process.c	2008-04-18 02:10:55.000000000 +0200
@@ -3785,7 +3785,7 @@
 	  free(blankerprg);
 	  blankerprg = 0;
 	}
-      if (args[0][0])
+      if (args[0] && args[0][0])
 	blankerprg = SaveArgs(args);
       break;
 #endif
